CMakePresets.json

```diff
@@ -0,0 +1,87 @@
+{
+  "version": 5,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 23,
+    "patch": 0
+  },
+  "configurePresets": [
+    {
+      "name": "default",
+      "displayName": "default",
+      "description": "Default preset that are inherited by all",
+      "generator": "Ninja",
+      "hidden": true,
+      "binaryDir": "${sourceDir}/../build_MyGodotProject"
+    },
+    {
+      "name": "default-windows",
+      "displayName": "64bit Windows",
+      "description": "Windows only!",
+      "inherits": "default",
+      "condition": {
+        "type": "equals",
+        "lhs": "${hostSystemName}",
+        "rhs": "Windows"
+      },
+      "toolset": {
+        "value": "host=x64",
+        "strategy": "external"
+      },
+      "architecture": {
+        "value": "x64",
+        "strategy": "external"
+      },
+      "cacheVariables": {
+        "CMAKE_C_COMPILER": "cl.exe",
+        "CMAKE_CXX_COMPILER": "cl.exe"
+      }
+    },
+    {
+      "name": "linux-debug",
+      "displayName": "64bit Debug Linux",
+      "description": "Linux only!",
+      "generator": "Ninja",
+      "condition": {
+        "type": "equals",
+        "lhs": "${hostSystemName}",
+        "rhs": "Linux"
+      },
+      "cacheVariables": {
+        "CMAKE_CXX_COMPILER": "g++",
+        "CMAKE_C_COMPILER": "gcc",
+        "CMAKE_BUILD_TYPE": "Debug"
+      }
+    },
+    {
+      "name": "osx-debug",
+      "displayName": "64bit Debug osx",
+      "description": "Osx only!",
+      "generator": "Ninja",
+      "condition": {
+        "type": "equals",
+        "lhs": "${hostSystemName}",
+        "rhs": "Darwin"
+      },
+      "cacheVariables": {
+        "CMAKE_CXX_COMPILER": "clang++",
+        "CMAKE_C_COMPILER": "clang",
+        "CMAKE_BUILD_TYPE": "Debug"
+      }
+    }
+  ],
+  "buildPresets": [
+    {
+      "name": "linux-debug",
+      "configurePreset": "linux-debug"
+    },
+    {
+      "name": "default-windows",
+      "configurePreset": "default-windows"
+    },
+    {
+      "name": "osx-debug",
+      "configurePreset": "osx-debug"
+    }
+  ]
```

Comment:
So does this only cover Linux & macOS debug, and Windows release? What about the Windows debug and Linux & macOS release?

Reply:
Ideally we should use [Ninja Multi Config](https://cmake.org/cmake/help/latest/generator/Ninja%20Multi-Config.html) but this did break so much in my project in therms of output paths that I reverted back. This multi config we no longer would have to hardcode release/debug/debugwithrelease versions

Reply:
Yes I'm aware of the multi-config issue - see https://github.com/asmaloney/GDExtensionTemplate/issues/25.

This is because godot-cpp's CMake is broken. I tried [to fix it](https://github.com/godotengine/godot-cpp/pull/957), but they don't seem to accept CMake changes very much and its way too much energy to keep trying.

Reply:
Ok then let keep it for now 👍